hwcpipe.cpp

```diff
@@ -129,58 +131,50 @@ void HWCPipe::set_enabled_gpu_counters(GpuCounterSet counters)
 	}
 }
 
-void HWCPipe::run()
+std::pair<Measurements, bool> HWCPipe::sample()
 {
-	if (cpu_profiler_)
-	{
-		cpu_profiler_->run();
-	}
-	if (gpu_profiler_)
-	{
-		gpu_profiler_->run();
-	}
-}
+	std::pair<Measurements, bool> result;
+	result.second = true;
 
-Measurements HWCPipe::sample()
-{
-	Measurements m;
 	if (cpu_profiler_)
 	{
-		m.cpu = &cpu_profiler_->sample();
-	}
-	if (gpu_profiler_)
-	{
-		m.gpu = &gpu_profiler_->sample();
+		if (!cpu_profiler_->poll())
+		{
+			// Failed to poll new samples
+			result.second = false;
```

Comment:
This can be written in a simpler form:
`
if (result.second = cpu_profiler_->poll()) {
  result.first.cpu = &cpu_profiler_->sample();
}
`
the same should be applied to gpu profiler